### PR TITLE
Remove static monitor helper functions

### PR DESCRIPTION
**Pull request by davidwrighton**
- Replace them with using the normal monitor helpers
- Change CORINFO_HELP_GETSYNCFROMCLASSHANDLE to return an objec instead of the underlying lock object, implementation of the JIT helper can be shared with other Typehandle to RuntimeType scenarios
- This will likely improve the performance of synchronized methods, but that is beside the point, since this feature is so rarely used

### CODE REVIEW

File: src/coreclr/tools/aot/ILCompiler.Compiler/IL/ILImporter.Scanner.cs [hunk 1]
```diff
@@ -151,8 +151,9 @@ public ILImporter(ILScanner compilation, MethodDesc method, MethodIL methodIL =
                 const string reason = "Synchronized method";
                 if (_canonMethod.Signature.IsStatic)
                 {
-                    _dependencies.Add(GetHelperEntrypoint(ReadyToRunHelper.MonitorEnterStatic), reason);
-                    _dependencies.Add(GetHelperEntrypoint(ReadyToRunHelper.MonitorExitStatic), reason);
+                    _dependencies.Add(GetHelperEntrypoint(ReadyToRunHelper.MonitorEnter), reason);
+                    _dependencies.Add(GetHelperEntrypoint(ReadyToRunHelper.MonitorExit), reason);
```
Comment (MichalStrehovsky) [inline src/coreclr/tools/aot/ILCompiler.Compiler/IL/ILImporter.Scanner.cs:155]: This dependency is now same between instance and static methods - can we avoid duplicating these lines?

**Review by MichalStrehovsky — COMMENTED** [review]
NativeAOT part LGTM otherwise, thanks for the simplification!

File: src/coreclr/tools/aot/ILCompiler.Compiler/IL/ILImporter.Scanner.cs (hunk 1 above)
Comment (davidwrighton) [reply src/coreclr/tools/aot/ILCompiler.Compiler/IL/ILImporter.Scanner.cs]: Done.

**Comment by JulieLeeMSFT** [comment]
@jakobbotsch, PTAL.

**Review by jakobbotsch — APPROVED** [review]

**Pull request merged**